### PR TITLE
Move minimum D3 version to 3.4.4

### PR DESCRIPTION
**Pull request by portante**
As of D3 commit:

https://github.com/mbostock/d3/commit/3c7cc81b6233289263525b581d635ad9f8d9dd37

We are no longer able to use an array accessor that accepts two
parameters (the second being the index of the first parameter in the
given array), since d3.bisector() will consider it a comparison
function instead of an accessor.

There does not appear to be a way to get the index value of the element
being considered in the context of the accessor.

As a result, this commit proposes we drop the interactiveBisect()
support for accessors accepting a second parameter being the index of
the first in a given array.

We now wrap all x-accessors passed to interactiveBisect() hiding the
second parameter, using an internal comparator instead.

This commit also attempts to get the coverage to 100% for
interactiveBisect().

See issue #754 as well.

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -1,6 +1,5 @@
 {
   "name": "nvd3",
-  "version": "1.7.1",
```
Comment (liquidpele) [inline bower.json]: version should stay there, the gruntfile reads it from here!

Comment (portante) [inline bower.json]: But I thought bower version numbers don't mean anything.  See https://github.com/bower/bower.json-spec

**Comment by robinfhu** [comment]
I'm okay with this actually.

**Comment by robinfhu** [comment]
Originally I wanted to find a way to support d3 3.3 and below, but now I think it's not worth it.

File: bower.json (hunk 1 above)
Comment (liquidpele) [inline bower.json]: They don't, but the gruntfile reads in the bower json to get the version, so the version is stored only in a single place :)

**Comment by liquidpele** [comment]
I'm fine with the D3 version bump, people should use close to latest anyway.  However, you should probably add a comment explanation in there to help when we re-read that code a year from now.

File: bower.json (hunk 1 above)
Comment (portante) [inline bower.json]: It looks to me like it reads package.json, no?

Comment (liquidpele) [inline bower.json]: Agh... In my defense I have not had my coffee yet!

**Comment by robinfhu** [comment]
Okay, so there's one small hiccup with this change.  In our code, we actually use function(d,i){return i;} for our accessor a lot.  The original d3 bisect code passed in the 'index' to the accessor function.  I need a way to get that index value.

**Comment by joshjordan** [comment]
Please also update the `README` with this change.

File: bower.json (hunk 1 above)
Comment (liquidpele) [inline bower.json]: :coffee: :coffee: :coffee:

**Comment by robinfhu** [comment]
I am going to try something slightly controversial.  I'm going to just write my own bisect left function.

**Comment by liquidpele** [comment]
So actually, I'm a little confused about this one anyway.  What exactly wasn't working?  I was doing my testing with d3 3.5 and never saw any issues.

**Comment by portante** [comment]
@robinfhu, is that really worth it?  What do we gain?

@joshjordan, okay, will do.

File: bower.json (hunk 1 above)
Comment (portante) [inline bower.json]: No defense necessary, all this stuff is quite confusing.

**Comment by robinfhu** [comment]
@portante @liquidpele , thought about it some more, I think you are right.  In some of our code, we've been relying on using the data index for the x-accessor.  We do function(d,i){ return i;} a lot.

But I realize now we can modify our code to manually add an index to each piece of data, so I can do something like: function(d,i){ return d._myIndex;}

So basically, I think everything is fine :-)

**Comment by robinfhu** [comment]
Thanks @portante

**Comment by Resseguie** [comment]
There were still a few version references out of sync after this merge. See #789

**Comment by liquidpele** [comment]
I merged the fix.

**Pull request merged**